src/cli/repl.rs

```diff
@@ -160,74 +150,92 @@ impl Repl<F> {
-                    let expression = self.store.get_z_expr(&input.expr, &mut zstore)?.0;
-                    let environment = self.store.get_z_expr(&input.env, &mut zstore)?.0;
-                    let result = self.store.get_z_expr(&output.expr, &mut zstore)?.0;
-
-                    info!("Proving and compressing");
-                    let start = Instant::now();
-                    let (proof, public_inputs, public_outputs, num_steps) =
-                        prover.prove(&pp, frames, &mut self.store, self.lang.clone())?;
-                    let generation = Instant::now();
-                    let proof = proof.compress(&pp)?;
-                    let compression = Instant::now();
-                    assert_eq!(self.rc * num_steps, n_frames);
-                    assert!(proof.verify(&pp, num_steps, &public_inputs, &public_outputs)?);
-
-                    let lurk_proof = &LurkProof::Nova {
-                        proof,
-                        public_inputs,
-                        public_outputs,
-                        num_steps,
-                        rc: self.rc,
-                        lang: (*self.lang).clone(),
-                    };
-
-                    let lurk_proof_meta = &LurkProofMeta {
-                        iterations: *iterations,
-                        evaluation_cost: *cost,
-                        generation_cost: generation.duration_since(start).as_nanos(),
-                        compression_cost: compression.duration_since(generation).as_nanos(),
-                        status,
-                        expression,
-                        environment,
-                        result,
-                        zstore: zstore.unwrap(),
-                    };
-
-                    let id = &format!("{}", timestamp());
-                    lurk_proof.persist(id)?;
-                    lurk_proof_meta.persist(id)?;
-                    println!("Proof ID: \"{id}\"");
+                    let expr = self.store.get_z_expr(&input.expr, &mut zstore)?.0;
+                    let env = self.store.get_z_expr(&input.env, &mut zstore)?.0;
+                    let cont = self.store.get_z_cont(&input.cont, &mut zstore)?.0;
+                    let expr_out = self.store.get_z_expr(&output.expr, &mut zstore)?.0;
+                    let env_out = self.store.get_z_expr(&output.env, &mut zstore)?.0;
+                    let cont_out = self.store.get_z_cont(&output.cont, &mut zstore)?.0;
+
+                    let proof_id = [
+                        expr.parts(),
+                        env.parts(),
+                        cont.parts(),
+                        expr_out.parts(),
+                        env_out.parts(),
+                        cont_out.parts(),
+                    ]
+                    .iter()
+                    .fold(F::ZERO, |acc, (a, b)| {
+                        self.store.poseidon_cache.hash3(&[acc, *a, *b])
+                    })
+                    .hex_digits();
```

Comment:
At this point, you could represent the claim as Lurk data — apart from the wrinkle around continuations not being expressions. Right?

Reply:
I think so

Reply:
It would be abstractly better to use Lurk data than to just hijack a random poseidon hash. That will get us closer to proofs about proofs, etc. and just generally keep us in the Lurk data ecosystem we've built.

Reply:
You can use `Store::list()`, I think. As a temporary hack, you could break the continuation into two `Num`s.

Reply:
`(:expr expr :env env :cont cont :expr-out expr-out :env-out env-out :cont-out cont-out)` might work.

Reply:
Or an alist, `((:expr . expr) (:env . env) …)`.

Reply:
Done